### PR TITLE
feat: Model additions for payTM integration

### LINKED ISSUE #6248: Addition of API keys for PayTM
**Feature Description**
Parent issue #6247 
Sub issue:
PayTM API keys require fields in the settings model.  Corresponding front-end fields will be added after this.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6248 

#### Short description of what this resolves:
PayTM uses a merchant key & a secret key to authorize payments.  These will be used in the follow-up PRs for the integration of the payment gateway. 

#### Changes proposed in this pull request:

- Model & schema additions PayTM
- Added hybrid properties to check for activation

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/schema/settings.py [hunk 1]
```diff
@@ -164,6 +164,8 @@ class Meta:
 
     # Stripe secret key
     stripe_secret_key = fields.Str(allow_none=True)
+    stripe_client_id = fields.Str(allow_none=True)
```
Comment (kushthedude) [inline app/api/schema/settings.py]: @mrsaicharan1 Why this change, arent this already there?

File: app/api/schema/settings.py [hunk 2]
```diff
@@ -164,6 +164,8 @@ class Meta:
...
+    stripe_publishable_key = fields.Str(allow_none=True)
```
Comment (kushthedude) [inline app/api/schema/settings.py]: @mrsaicharan1 Why this change, arent this already there?

File: app/models/setting.py [hunk 3]
```diff
@@ -303,6 +315,13 @@ def __init__(self,
         self.alipay_publishable_key = alipay_publishable_key
         self.alipay_secret_key = alipay_secret_key
 
+        # Omise Credentials
```
Comment (kushthedude) [inline app/models/setting.py]: PayTM Credantials !

File: app/models/setting.py [hunk 4]
```diff
@@ -356,3 +375,12 @@ def is_omise_activated(self):
             return True
         else:
             return False
+
+    @hybrid_property
+    def is_paytm_activated(self):
```
Comment (kushthedude) [inline app/models/setting.py]: Why didn't column you add a boolean for this?

Comment (prateekj117) [reply app/models/setting.py]: @kushthedude Because it's a violation of [Third Normal Form](https://en.wikipedia.org/wiki/Third_normal_form).

Comment (iamareebjamal) [reply app/models/setting.py]: No, it isn't. What if I want to add PayTM API keys but not activate PayTM. Or I have added the keys and everything, but I want to disable PayTM. Now I can't do that unless I remove one of the keys, which seems unnecessary

Comment (prateekj117) [reply app/models/setting.py]: @iamareebjamal Yaa, it won't be in that case.

File: app/api/schema/settings.py (hunk 2 above)
Comment (mrsaicharan1) [reply app/api/schema/settings.py]: @kushthedude This wasn't required. I will change this.

**Comment by mrsaicharan1** [comment]
@iamareebjamal @uds5501 @shreyanshdwivedi Please have a look

File: app/models/setting.py (hunk 4 above)
Comment (kushthedude) [reply app/models/setting.py]: @mrsaicharan1 I think you missed it

Comment (mrsaicharan1) [reply app/models/setting.py]: Okay, I will make it a boolean field.

File: app/api/schema/settings.py [hunk 5]
```diff
@@ -185,6 +185,17 @@ class Meta:
     alipay_publishable_key = fields.Str(allow_none=True)
     alipay_secret_key = fields.Str(allow_none=True)
 
+<<<<<<< Updated upstream
+=======
+    # payTM credentials
+
+    is_paytm_activated = fields.Bool(allow_none=False, default=False)
```
[the remark this answers is not in the document]
Comment (uds5501) [reply app/api/schema/settings.py]: declare `is_paytm_activated` in `SettingSchemaPublic` in similar manner as others.

[the remark this answers is not in the document]
Comment (mrsaicharan1) [reply app/api/schema/settings.py]: @uds5501 Done.

File: app/models/setting.py [hunk 6]
```diff
@@ -361,3 +382,12 @@ def is_omise_activated(self):
...
+        if self.paytm_mode == 'test' and self.paytm_sandbox_merhcant and self.paytm_sandbox_secret:
```
Comment (kushthedude) [inline app/models/setting.py]: Remove this block

File: app/api/schema/settings.py [hunk 7]
```diff
@@ -58,6 +58,7 @@ class Meta:
     #
     # Online Payment Flags
     #
+    is_paytm_activated = fields.Bool(allow_none=False, default=False)
```
Comment (kushthedude) [inline app/api/schema/settings.py]: `Dump_only` ?

**Comment by mrsaicharan1** [comment]
```
SAWarning: Attribute 'is_paytm_activated' on class <class 'app.models.setting.Setting'> appears to be a non-schema 'sqlalchemy.sql.column()' object; this won't be part of the declarative mapping
  % (key, cls)
```
@kushthedude @iamareebjamal I'm getting this warning when I restart the server. Aren't the model and schema fields present here?

**Comment by mrsaicharan1** [comment]
> ```
> SAWarning: Attribute 'is_paytm_activated' on class <class 'app.models.setting.Setting'> appears to be a non-schema 'sqlalchemy.sql.column()' object; this won't be part of the declarative mapping
>   % (key, cls)
> ```
> 
> @kushthedude @iamareebjamal I'm getting this warning when I restart the server. Aren't the model and schema fields present here?

Fixed

File: app/models/setting.py [hunk 8]
```diff
@@ -222,6 +230,12 @@ def __init__(self,
                  omise_live_secret=None,
                  alipay_publishable_key=None,
                  alipay_secret_key=None,
+                 is_paytm_activated=None,
```
Comment (uds5501) [inline app/models/setting.py]: this should be `is_paytm_activated=False`

Comment (mrsaicharan1) [reply app/models/setting.py]: Taken care of in the recent iteration.

**Comment by mrsaicharan1** [comment]
Travis wasn't able to start. I've rebase and pushed once again. Please review. @kushthedude @uds5501 @shreyanshdwivedi
